### PR TITLE
PLANNER-936: Add OptaPlanner's code formatter to the backend

### PR DESCRIPTION
**Pull request by Christopher-Chianelli**
Only changes in this PR are:
- adding automatic format on compile of back end (copied from optaplanner-core)
- any formatting changes from said format

### CODE REVIEW

File: optaweb-employee-rostering-backend/pom.xml
Comment (triceo) [inline optaweb-employee-rostering-backend/pom.xml:277]: I'd remove this, just like we removed it in core.

Comment (Christopher-Chianelli) [reply optaweb-employee-rostering-backend/pom.xml:277]: It not removed in core; that the only reason why I kept it.
`optaplanner-build-parent/pom.xml`:
```
      <plugin>
        <artifactId>maven-checkstyle-plugin</artifactId>
      </plugin>
```

Comment (triceo) [reply optaweb-employee-rostering-backend/pom.xml:277]: Interesting, must have missed that one when I was removing the checkstyle config. Will finish it. :-)

File: optaweb-employee-rostering-backend/pom.xml
Comment (yurloc) [inline optaweb-employee-rostering-backend/pom.xml:243]: 1. The plugin configuration should go to the parent's `<pluginManagement>` so that it can be reused in child modules.
2. The plugin should be added to the build (`<build>/<plugins>`) in modules that contain Java source code (backend, benchmark).

**Review by yurloc — CHANGES_REQUESTED** [review]
The formatter and impsort plugins should run in the benchmark module as well. Otherwise looks good.

I would recommend to merge this after finishing the work in progress. It will be easier to rebase this PR, drop conflicting changes and rerun formatter than the other way around.

File: optaweb-employee-rostering-backend/pom.xml
Comment (yurloc) [inline optaweb-employee-rostering-backend/pom.xml:288]: One last thing. This profile should go to the parent as well so that the build fails if formatting is broken in benchmark or any other module.

File: optaweb-employee-rostering-backend/pom.xml
Comment (yurloc) [inline optaweb-employee-rostering-backend/pom.xml:287]: Trailing whitespace. Would be nice to get rid of that before merging the PR.

**Pull request merged**
